catt/controllers.py

```diff
@@ -95,11 +94,10 @@ def get_stream(url, device_info=None, host=None, ytdl_options=None):
 
 
 def get_app(id_or_name, cast_type=None, strict=False, show_warning=False):
-    arg_is_id = True if re.match("[0-9A-F]{8}$", id_or_name) else False
     try:
-        app = next(a for a in APPS if (a.id if arg_is_id else a.name) == id_or_name)
+        app = next(a for a in APPS if id_or_name in [a.id, a.name])
     except StopIteration:
-        if strict:
+        if strict and id_or_name != "default":
```

Comment:
The simplification of logic in the previous PR introduced a bug. I have a test (as featured in ```test_procedure.py```) that casts low resolution content from twitch using ```--ytdl-opt```. In this case ```strict``` would be set, so we would fail, as "default" is being specifically requested.

Reply:
Ah, I see. Is there a reason why we aren't including the default app in the app list?

Also, I guess this is why you had the short-circuit in the logic, but there was no accompanying comment, so we forgot about the reason. That's a lesson in always leaving comments for non-obvious functionality.

Reply:
It looks like we only use `APPS` in `get_app`, but then special-case `DEFAULT_APP` by not including it in the list. The code would be simpler if we did, no?

Reply:
> That's a lesson in always leaving comments for non-obvious functionality

Noted. :smiley:

Reply:
> It looks like we only use APPS in get_app, but then special-case DEFAULT_APP by not including it in the list. The code would be simpler if we did, no?

True. The problem is that we are using ```DEFAULT_APP``` later in ```get_app``` and then also in [```DefaultCastController```](https://github.com/skorokithakis/catt/blob/master/catt/controllers.py#L535).
We could of course have our cake and eat it by including the default app in ```APPS```, and then do ```DEFAULT_APP = next(a for a in APPS if a.name ==  "default")```.

Reply:
(or we could change the datastructure again :smile: )